src/main/scala/dpla/ingestion3/harvesters/resourceSync/RsResponseBuilder.scala

```diff
@@ -0,0 +1,169 @@
+package dpla.ingestion3.harvesters.resourceSync
+
+import java.net.{URI, URL}
+
+import org.apache.commons.io.IOUtils
+import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet}
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SQLContext
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+import scala.xml.XML
+
+
+/**
+  * This class handles requests to the ResourceSync feed.
+  * It partitions data per resourceList.
+  */
+
+class RsResponseBuilder (endpoint: String)
+                         (@transient val sqlContext: SQLContext) extends Serializable {
+  /**
+    * Gets all the records described by the source
+    *
+    * @return RDD[RsResponse]
+    */
+  def getAllResourceList(): RDD[RsResponse] = {
+    val multiPageResponse = getMultiResourceResponse()
+    sqlContext.sparkContext.parallelize(multiPageResponse)
+  }
+
+  /**
+    * Expects the capabilitylist list URL and returns a list of the capabilities of the endpoint
+    *
+    * @see https://www.openarchives.org/rs/1.1/resourcesync#CapabilityList
+    * @param capUrl URL that describes the capabilities of the source
+    * @return Map of the capability term and it's corresponding URL
+    */
+  def getCapibilityUrls(capUrl: URL): Map[String, String] = {
+    val rsp = XML.load(capUrl)
+    (rsp \\ "url").map( u => {
+      (u \\ "@capability").text -> (u \\ "loc").text
+    }).toMap
+  }
+
+  /**
+    * Take the capability URL and extracts all of the resource lists
+    * defined
+    *
+    * TODO This should support Sitemaps in addition to RS capabilities
+    * @see https://www.openarchives.org/rs/1.1/resourcesync#Walkthrough
+    *
+    * @param  url Capability URL
+    * @return Sequence of resource list urls
+    */
+  def getResourceLists(url: URL): Seq[String] = {
+    val rsp = XML.load(url)
+    (rsp \\ "url").map( u => {
+      (u \\ "loc").text -> (u \\ "@capability").text
+    }).filter(pairs => pairs match {
+      case (_, "resourcelist") => true
+      case (_,_) => false
+    }).map(r => r._1)
```

Comment:
I think here you could do something like `.map{ case (r, _) => r }` to make codacy happy.

Reply:
Thanks for the suggestion. I'll take a stab.

Reply:
Fixed in the last commit. Thanks!